### DISCUSSION TITLE
v0.6.0-RC1

### DISCUSSION THREAD

**Discussion by erikerlandson** [erikerlandson/coulomb#329] [General]
- The first release candidate for coulomb on Scala 3
- laika/mdoc documentation is completed for coulomb-core, coulomb-units, coulomb-spire
- `scala3` is the new default/main branch for the `coulomb` repository
- ghpages publishing is now enabled for `scala3` branch
- thanks to @armanbilge and @cquiroz for all of their help with modernizing coulomb for scala 3!

<hr /><em>This discussion was created from the release <a href='https://github.com/erikerlandson/coulomb/releases/tag/v0.6.0-RC1'>v0.6.0-RC1</a>.</em>

**Comment by erikerlandson** [comment]
@cquiroz @armanbilge 
I have cut the first official release candidate for coulomb on scala 3 :tada: 

So, now is a good time to file any last issues, prs, make suggestions, etc.

@armanbilge the two things on my mind for the leap to official release are:
- enabling mima
- enabling scalafmt

I suspect the first time I turn on scalafmt, it will have a *lot* to say about changing the existing code.  I'm guessing mima may actually be easier until the next time someone tries to make a breaking change.  If anyone has thoughts about making changes that might be breaking, now is a good time to raise them, I guess.

**Reply by armanbilge** [reply]
@erikerlandson congratulations! We've been very excited for a final release :) the website and documentation look great!

1. MiMa is already enabled. As soon as you publish your final release 0.6.0 it will kick-in automatically for all subsequent PRs.
2. You can enable scalafmt by removing this line.
https://github.com/erikerlandson/coulomb/blob/31e44f73ae75a386f3b4cf5315f0c7c633b846ca/.scalafmt.conf#L3

    You are right, I expect it will want to make many changes. I guess you can explore different configuration options to see what best fits your style and run `scalafmtAll` to assess the damage 😅 

    https://scalameta.org/scalafmt/docs/configuration.html

    Fortunately, you can hide the formatting commit from `git blame` using a `.git-blame-ignore-revs` file:
https://github.com/typelevel/spire/blob/9843b770f41fb91d2c44c9cda8cb7f225ae3706a/.git-blame-ignore-revs

    Or ... you can just leave it disabled too :) up to you, not every likes auto-formatting.
